src/sites/twitch-twilight/modules/directory/following.js

```diff
@@ -194,34 +207,46 @@ export default class Following extends SiteModule {
 		return res;
 	}
 
+	ensureQueries () {
+		if (this.router && this.router.match) {
+			this.apollo.ensureQuery(
+				'FollowedChannels',
+				'data.currentUser.followedLiveUsers.nodes.0.profileImageURL'
+			);
+
+			if (this.router.match[1] === 'following') {
+				this.apollo.ensureQuery(
+					'FollowedIndex_CurrentUser',
+					n =>
+						get('data.currentUser.followedLiveUsers.nodes.0.profileImageURL', n) !== undefined
+						||
+						get('data.currentUser.followedHosts.nodes.0.hosting.profileImageURL', n) !== undefined
+				);
+			} else if (this.router.match[1] === 'live') {
+				this.apollo.ensureQuery(
+					'FollowingLive_CurrentUser',
+					'data.currentUser.followedLiveUsers.nodes.0.profileImageURL'
+				);
+			} else if (this.router.match[1] === 'hosts') {
+				this.apollo.ensureQuery(
+					'FollowingHosts_CurrentUser',
+					'data.currentUser.followedHosts.nodes.0.hosting.profileImageURL'
+				);
+			}
+		}
+	}
+
 	onEnable() {
 		this.ChannelCard.ready((cls, instances) => {
-			if (this.router && this.router.match) {
-				if (this.router.match[1] === 'following') {
-					this.apollo.ensureQuery(
-						'FollowedIndex_CurrentUser',
-						n =>
-							get('data.currentUser.followedLiveUsers.nodes.0.profileImageURL', n) !== undefined
-							||
-							get('data.currentUser.followedHosts.nodes.0.hosting.profileImageURL', n) !== undefined
-					);
-				} else if (this.router.match[1] === 'live') {
-					this.apollo.ensureQuery(
-						'FollowingLive_CurrentUser',
-						'data.currentUser.followedLiveUsers.nodes.0.profileImageURL'
-					);
-				} else if (this.router.match[1] === 'hosts') {
-					this.apollo.ensureQuery(
-						'FollowingHosts_CurrentUser',
-						'data.currentUser.followedHosts.nodes.0.hosting.profileImageURL'
-					);
-				}
-			}
+			this.ensureQueries();
 
 			for(const inst of instances) this.updateChannelCard(inst);
 		});
 
-		this.ChannelCard.on('update', inst => this.updateChannelCard(inst), this);
+		this.ChannelCard.on('update', inst => {
+			this.ensureQueries();
+			this.updateChannelCard(inst)
+		}, this);
 		this.ChannelCard.on('mount', inst => this.updateChannelCard(inst), this);
```

Comment:
You don't need an arrow function here. The point of the third argument to `.on(...)` is that you're providing a context that the method is called with.